test/tests/functional/pbs_fairshare.py

```diff
@@ -206,6 +213,86 @@ def test_fairshare_formula4(self):
         for i in range(len(job_order)):
             self.assertEqual(job_order[i].split('.')[0], c.political_order[i])
 
+    def test_fairshare_formula5(self):
+        """
+        Test fairshare in the formula with fair_share set to true in scheduler.
+        Make sure formula takes precedence over fairshare usage. Output will be
+        same as in test_fairshare_formula4. 
+        """
+
+        self.set_up_resource_group()
+        a = {'log_filter' : 2048, 'fair_share': "True All"}
+        self.scheduler.set_sched_config(a)
+
+        formula = 'fairshare_factor'
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'job_sort_formula': formula})
+
+        J1 = Job(TEST_USER2, {'Resource_List.cput': 10})
+        jid1 = self.server.submit(J1)
+        J2 = Job(TEST_USER3, {'Resource_List.cput': 20})
+        jid2 = self.server.submit(J2)
+        J3 = Job(TEST_USER, {'Resource_List.cput': 30})
+        jid3 = self.server.submit(J3)
+        J4 = Job(TEST_USER1, {'Resource_List.cput': 40})
+        jid4 = self.server.submit(J4)
+        t = int(time.time())
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        msg = ';Formula Evaluation = '
+        self.scheduler.log_match(str(jid1) + msg + '0.3816', max_attempts=2)
+        self.scheduler.log_match(str(jid2) + msg + '0.0902', max_attempts=2)
+        self.scheduler.log_match(str(jid3) + msg + '0.6477', max_attempts=2)
+        self.scheduler.log_match(str(jid4) + msg + '0.6477', max_attempts=2)
+        self.scheduler.log_match('Leaving Scheduling Cycle', starttime=t)
+
+        c = self.scheduler.cycles(start=t, lastN=1)[0]
+        job_order = [jid3, jid4, jid1, jid2]
+        for i in range(len(job_order)):
+            self.assertEqual(job_order[i].split('.')[0], c.political_order[i])
```

Comment:
I guess my comment wasn't clear.  My point you'd get the same result if you tested fairshare_factor or fair_share.  Since you're testing that the job sort formula overrides fair_share, you won't actually know since they both give the same job ordering.